pkg/controllers/execution/execution_controller.go

```diff
@@ -133,56 +133,37 @@ func (c *Controller) syncWork(clusterName string, work *workv1alpha1.Work) (cont
 	err := c.syncToClusters(clusterName, work)
 	metrics.ObserveSyncWorkloadLatency(err, start)
 	if err != nil {
-		msg := fmt.Sprintf("Failed to sync work(%s) to cluster(%s): %v", work.Name, clusterName, err)
+		msg := fmt.Sprintf("Failed to sync work(%s/%s) to cluster(%s), err: %v", work.Namespace, work.Name, clusterName, err)
 		klog.Errorf(msg)
 		c.EventRecorder.Event(work, corev1.EventTypeWarning, events.EventReasonSyncWorkloadFailed, msg)
 		return controllerruntime.Result{}, err
 	}
-	msg := fmt.Sprintf("Sync work (%s) to cluster(%s) successful.", work.Name, clusterName)
+	msg := fmt.Sprintf("Sync work(%s/%s) to cluster(%s) successful.", work.Namespace, work.Name, clusterName)
 	klog.V(4).Infof(msg)
 	c.EventRecorder.Event(work, corev1.EventTypeNormal, events.EventReasonSyncWorkloadSucceed, msg)
 	return controllerruntime.Result{}, nil
 }
 
-// tryDeleteWorkload tries to delete resource in the given member cluster.
+// tryDeleteWorkload tries to delete resources in the given member cluster.
 func (c *Controller) tryDeleteWorkload(clusterName string, work *workv1alpha1.Work) error {
+	var errs []error
 	for _, manifest := range work.Spec.Workload.Manifests {
 		workload := &unstructured.Unstructured{}
 		err := workload.UnmarshalJSON(manifest.Raw)
 		if err != nil {
-			klog.Errorf("Failed to unmarshal workload, error is: %v", err)
-			return err
-		}
-
-		fedKey, err := keys.FederatedKeyFunc(clusterName, workload)
-		if err != nil {
-			klog.Errorf("Failed to get FederatedKey %s, error: %v", workload.GetName(), err)
-			return err
-		}
-
-		clusterObj, err := helper.GetObjectFromCache(c.RESTMapper, c.InformerManager, fedKey)
-		if err != nil {
-			if apierrors.IsNotFound(err) {
-				return nil
-			}
-			klog.Errorf("Failed to get resource %v from member cluster, err is %v ", workload.GetName(), err)
-			return err
-		}
-
-		// Avoid deleting resources that not managed by karmada.
-		if util.GetLabelValue(clusterObj.GetLabels(), util.ManagedByKarmadaLabel) != util.ManagedByKarmadaLabelValue {
```

Comment:
I'm a little confused why you moved this logic to the ObjectWatcher.Delete function.

Reply:
I think judging whether it can be deleted should be the responsibility of the `ObjectWatcher.Delete` function. We should not judge it outside of calling it. Just like the `ObjectWatcher.Update` function, it will also check whether it can be updated.
In addition, if the `ObjectWatcher.Delete` function is called elsewhere in the future, this part of the logic can be reused to the greatest extent.

Reply:
Ok, I get it.

Reply:
But we can do that in a separated PR.